File: GMS-backend/GMS/Entities/Endpoint/Korisnik_Suplement/GetAll/Korisnik_SuplementGetAllEndpoint.cs
using GMS.Data;
using GMS.Helpers;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace GMS.Entities.Endpoint.KorisnikSuplement.GetAll
{
    [Route("Korisnik_Suplement-GetAll")]

    public class Korisnik_SuplementGetAllEndpoint : MyBaseEndpoint<Korisnik_SuplementGetAllRequest, Korisnik_SuplementGetAllResponse>
    {
        private readonly ApplicationDbContext db;

        public Korisnik_SuplementGetAllEndpoint(ApplicationDbContext db)
        {
            this.db = db;
        }

        [HttpGet]
        public override async Task<Korisnik_SuplementGetAllResponse> Handle([FromQuery] Korisnik_SuplementGetAllRequest request, CancellationToken cancellationToken)
        {
            var korisniksuplement = await db.Korisnik_Suplement
                .Select(x => new KorisnikSuplementGetAllResponseRow
                {
                    SuplementID = x.SuplementID,
                    KorisnikID = x.KorisnikID,
                    DatumVrijemeNarudzbe = x.DatumVrijemeNarudzbe,
                    Kolicina = x.Kolicina,
                    Cijena = x.Suplement.Cijena,
                    ImePrezime = x.Korisnik.Ime + " " + x.Korisnik.Prezime,
                    NazivSuplementa = x.Suplement.Naziv,
                    Ukupno = x.Suplement.Cijena * x.Kolicina,
                    Isporuceno = x.Isporuceno


                }).OrderByDescending(x => x.DatumVrijemeNarudzbe).ToListAsync(cancellationToken: cancellationToken);

            return new Korisnik_SuplementGetAllResponse
            {
                KorisnikSuplement = korisniksuplement
            };
        }
    }
}

File: GMS-frontend/src/app/endpoints/korisnik-endpoints/korisnik-suplementi-endpoints/korisnik-suplement-getall-endpoint.ts

import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Injectable} from "@angular/core";
import {Config} from "../../../config";
import {MyBaseEndpoint} from "../../MyBaseEndpoint";
@Injectable({providedIn: 'root'})
export class Korisnik_SuplementGetAllEndpoint implements MyBaseEndpoint<void, Korisnik_SuplementGetAllResponse>{
  constructor(public httpClient:HttpClient) { }

  Handle(request: void): Observable<Korisnik_SuplementGetAllResponse> {
    let url = Config.adresa + 'Korisnik_Suplement-GetAll';

    let token = window.localStorage.getItem("my-auth-token")??"";

    return this.httpClient.get<Korisnik_SuplementGetAllResponse>(url , {
      headers:{
        "my-auth-token": token
      }
    });
  }
}

export interface Korisnik_SuplementGetAllResponse{
  korisnikSuplement : Korisnik_SuplementGetAllResponseKorisnik_Suplement[];

}

export interface Korisnik_SuplementGetAllResponseKorisnik_Suplement {
  korisnikID: number
  imePrezime: string
  suplementID: number
  nazivSuplementa: string
  datumVrijemeNarudzbe: string
  kolicina: number
  cijena: number
  ukupno: number
  isporuceno: boolean
}
